tests/test_processors.py

```diff
@@ -321,3 +331,76 @@ def aggregateRating(self):
     assert page.aggregateRating == AggregateRating(
         ratingValue=3.8, bestRating=10, reviewCount=5
     )
+
+
+@pytest.mark.parametrize(
+    "input_value,expected_value",
+    [
+        (None, None),
+        ([], []),
+        ("https://www.url.com/img.jpg", [Image(url="https://www.url.com/img.jpg")]),
+        (
+            [
+                Image("https://www.url.com/img1.jpg"),
+                Image("https://www.url.com/img2.jpg"),
+            ],
+            [
+                Image("https://www.url.com/img1.jpg"),
+                Image("https://www.url.com/img2.jpg"),
+            ],
+        ),
+        (
+            ["https://www.url.com/img1.jpg", "https://www.url.com/img2.jpg"],
+            [
+                Image("https://www.url.com/img1.jpg"),
+                Image("https://www.url.com/img2.jpg"),
+            ],
+        ),
+        (
+            [
+                {"url": "https://www.url.com/img1.jpg"},
+                {"url": "https://www.url.com/img2.jpg"},
+            ],
+            [
+                Image("https://www.url.com/img1.jpg"),
+                Image("https://www.url.com/img2.jpg"),
+            ],
+        ),
+    ],
+)
+def test_images(input_value, expected_value):
+    class ImagesPage(BasePage):
+        @field(out=[images_processor])
+        def images(self):
+            return input_value
+
+    page = ImagesPage(base_url)  # type: ignore[arg-type]
+    assert page.images == expected_value
+
+
+@pytest.mark.parametrize(
+    "input_value,expected_value",
+    [
+        ("$10", "10.00"),
+        ("100  ", "100.00"),
+        ("100rub", "100.00"),
+        (100, "100.00"),
+        (None, None),
+        ([], []),
+        ({}, {}),
+        ("", None),
+        ("buy 10 ab ab", "10.00"),
+        ("1,000.17", "1000.17"),
+        ("1,000", "1000.00"),
+        (22.9, "22.90"),
+        (22.0, "22.00"),
+    ],
+)
+def test_prices(input_value, expected_value):
+    class PricePage(BasePage):
+        @field(out=[price_processor])
```

Comment:
is `out` needed?

Reply:
In this case yes, base class is `BasePage` so there's no default processor for `price` field and I need to mark it explicitly.